### PR TITLE
117 Fix rank de-serialization bug in Kokkos::DynRankView

### LINKED ISSUE #117: After de-serializing Kokkos::DynRankView, rank is always 7
Refer to: https://cee-gitlab.sandia.gov/EM-Plasma/EMPIRE/-/merge_requests/1127#note_500093

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #117

Refer to: https://cee-gitlab.sandia.gov/EM-Plasma/EMPIRE/-/merge_requests/1127#note_500093

### CODE REVIEW

**Comment by lifflander** [comment]
@PhilMiller Can you review?

**Review by PhilMiller — APPROVED** [review]

File: src/checkpoint/container/view_serialize.h [hunk 1]
```diff
@@ -264,12 +264,19 @@ inline void serialize_impl(SerializerT& s, Kokkos::DynRankView<T,Args...>& view)
...
+  // Make sure we serialize all 7 dimensions, instead of just `dims`.
   if (s.isUnpacking()) {
     serializeLayout<SerializerT>(s, 7, layout);
   } else {
     ArrayLayoutType layout_cur = view.layout();
+
+    // We must set these to the invalid index (they are not set in the layout by
+    // default!). This ensures that when the DynRankView comes out after
+    // de-serialization, the number of ranks is correct.
+    for (int i = dims; i < 8; i++) {
```
Comment (PhilMiller) [inline src/checkpoint/container/view_serialize.h:276]: I just realized - should this loop bound be 8, or 7? Kokkos Views allow up to 7 dimensions. Is there some weird 1-based indexing of `ArrayLayout`?

Comment (PhilMiller) [reply src/checkpoint/container/view_serialize.h:276]: Never mind, looks like 8 is right
https://github.com/kokkos/kokkos/wiki/Kokkos%3A%3ALayoutRight

Comment (PhilMiller) [reply src/checkpoint/container/view_serialize.h:276]: Ugh, documentation says View goes up to 8 dimensions, while DynRankView goes to 7
https://github.com/kokkos/kokkos/wiki/View
https://github.com/kokkos/kokkos/wiki/Kokkos%3A%3ADynRankView

**Pull request merged**